pass-data-client/src/main/java/org/dataconservancy/pass/client/PassClientDefault.java

```diff
@@ -84,6 +86,40 @@ public void deleteResource(URI modelObj) {
         return crudClient.getIncoming(passEntity);
     }
 
+    @Override
+    public URI upload(URI entityUri, InputStream content) {
+        return upload(entityUri, content, Collections.emptyMap());
+    }
+
+    /**
+     * {@inheritDoc}
+     * <h4>Implementation notes</h4>
+     * Supported parameters include:
+     * <dl>
+     *     <dt>content-type</dt>
+     *     <dd>the mime type of the {@code content} to be {@code POST}ed; added as a {@code Content-Type} header</dd>
+     *     <dt>slug</dt>
+     *     <dd>suggested name of the resource in the repository; added as a {@code Slug} header</dd>
+     *     <dt>sha256</dt>
+     *     <dd>hexadecimal encoded SHA-256 checksum of {@code content}; added as {@code Digest} header</dd>
+     *     <dt>sha1</dt>
+     *     <dd>hexadecimal encoded SHA-1 checksum of {@code content}; added as {@code Digest} header</dd>
+     *     <dt>md5</dt>
+     *     <dd>hexadecimal encoded MD5 checksum of {@code content}; added as {@code Digest} header</dd>
+     *     <dt>filename</dt>
```

Comment:
I'm torn about this.  On one hand, the supported parameters probably belongs on the implementation.  On the other hand, that means that a developer wanting to know what to put in that map has to (a) figure out what the underlying implementation is, and (b) then look at its doc.

For me, it would be clearer to put this  on the `PassClient` interface javadocs and make it part of the contract.   Stuff like `content-type ` is fairly fundamental.